api/index.py

```diff
@@ -184,6 +192,41 @@ def get_stats_response():
         )
     abort("Method not allowed", 405)
 
+@app.route("/compare")
+def compare():
+    if request.method == "GET":
+        try :
+            comp_type = json.loads(request.args.get("Type"))
+        except json.decoder.JSONDecodeError as json_decode_error:
+            raise InvalidParameterException(
+                "Possible values - 3d, 2d"
+            ) from json_decode_error
+        try:
+            years = year_input_manager(json.loads(request.args.get("Year")))
+        except json.decoder.JSONDecodeError as json_decode_error:
+            raise InvalidParameterException(
+                "The Year should either be a Number, array of number or a string of tuple"
+            ) from json_decode_error
+        try: 
+            _, countries = region_input_manager(json.loads(request.args.get("Region")))
+        except json.decoder.JSONDecodeError as json_decode_error:
+            raise InvalidParameterException(
+                "The Region should either be a string enclosed by quotation"
+            ) from json_decode_error
+        queryset_pop = Population.query.filter(
+            Population.year.in_(years), Population.country.in_(countries)
+        )
+        queryset_gdp = GDP_per_capita.query.filter(
+            GDP_per_capita.year.in_(years), GDP_per_capita.country.in_(countries)
+        )
+        json_response_pop = serialize_queryset(queryset_pop)
+        json_response_gdp = serialize_queryset_gdp(queryset_gdp)
```

Comment:
This needs to be changed. We shouldn't write two almost similar functions twice.